spring-cloud-vault-config-databases/src/main/java/org/springframework/cloud/vault/config/databases/VaultCouchbaseProperties.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2016-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.vault.config.databases;
+
+import javax.validation.constraints.NotEmpty;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.validation.annotation.Validated;
+
+/**
+ * Configuration properties for Vault using the Couchbase integration.
+ *
+ * @author Francis Hitchens
+ */
+@ConfigurationProperties("spring.cloud.vault.couchbase")
+@Validated
+public class VaultCouchbaseProperties implements DatabaseSecretProperties {
+
+	/**
+	 * Enable couchbase backend usage.
+	 */
+	private boolean enabled = false;
+
+	/**
+	 * Role name for credentials.
+	 */
+	private String role;
+
+	/**
+	 * Enable static role usage.
+	 *
+	 * @since 2.2
```

Comment:
Note: `@since` should go into the class documentation.

Reply:
Removed as static role has always been supported by the couchbase plugin